applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_law.cpp

```diff
@@ -54,10 +54,8 @@ void GeoLinearElasticLaw::CalculateMaterialResponsePK2(ConstitutiveLaw::Paramete
 
     Vector& r_strain_vector = rValues.GetStrainVector();
 
-    // NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
-    if (r_options.IsNot(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN)) {
-        CalculateCauchyGreenStrain(rValues, r_strain_vector);
-    }
+    KRATOS_ERROR_IF_NOT(r_options.Is(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN))
+        << "The GeoLinearElasticLaw needs an element provided strain" << std::endl;
```

Comment:
As the calling routine needs to provide the strain, it is also responsible for providing an appropriate strain. In other words, I don't expect this check here anymore.

Is there from line 55 a way to ensure that strain was provided and otherwise stop.

Reply:
Changed to check for validity of strain vector instead of the option, let me know what you think!

Reply:
I would suggest to keep the check (since option `USE_ELEMENT_PROVIDED_STRAIN` is now implicitly assumed; it cannot be set to `false`), but make it a debug error rather than a "normal" error:
```cpp
KRATOS_DEBUG_ERROR_IF_NOT(r_options.Is(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN))
    << "The GeoLinearElasticLaw needs an element provided strain\n";
```